Close overlay panels on click in body

MattIPv4 opened the pull request:
Resolves #61.

This is slightly hacky, in that it relies on checking the parent of the overlay itself, so that click on setting doesn't close the panels. It should work as desired though.

abdullahmorrison reviewed (changes requested):
Pressing the x on the top of the ambassador card to close it closes the entire ambassador list instead of just the card. It needs to only close the card.

Also, the overlay doesn't dismiss if you click above or below the card. I'm fine with it if the solution to this is overly complicated or a lot of work.

MattIPv4 commented:
Good feedback, ty!

The first one was due to when in the event loop this logic was being run -- I've changed it to the capture phase so we can process this before regular click handlers run.

As for the second, good shout, I've updated the logic to loop through the elements under the mouse to determine whether the user is clicking on the overlay or not. That should solve the issue you had, and is actually less jank that my prior approach for handling settings too!

abdullahmorrison reviewed (approved):
I would have never, in a million years, thought to do it this way. I like it!
